cmd/podman/libpodruntime/runtime.go

```diff
@@ -24,23 +25,23 @@ func GetRootlessStorageOpts() (storage.StoreOptions, error) {
 
 	opts.RunRoot = filepath.Join(libpod.GetRootlessRuntimeDir(), "run")
 
-	dataDir := os.Getenv("XDG_DATA_DIR")
+	dataDir := os.Getenv("XDG_DATA_HOME")
 	if dataDir != "" {
 		opts.GraphRoot = filepath.Join(dataDir, "containers", "storage")
 	} else {
 		home := os.Getenv("HOME")
 		if home == "" {
 			return opts, fmt.Errorf("HOME not specified")
 		}
-		opts.GraphRoot = filepath.Join(home, ".containers", "storage")
+		opts.GraphRoot = filepath.Join(home, ".local", "share", "containers", "storage")
```

Comment:
The [spec has][1]:

> If `$XDG_DATA_HOME` is either not set or empty, a default equal to `$HOME/.local/share` should be used.

A simpler implementation of that would be:

```go
dataDir := os.Getenv("XDG_DATA_HOME")
if dataDir == "" {
  home := os.Getenv("HOME")
  if home == "" {
    return opts, fmt.Errorf("neither XDG_DATA_HOME nor HOME was set non-empty")
  }
  dataDir = filepath.Join(home, ".local", "share")
}
// then append to dataDir to construct GraphRoot
```

This sort of generic XDG support is something that you can get from XDG libraries as well.  For example, [here][2].

Once you have `dataDir` populated, I think we should be using a single `containers-storage` instead of two directories.  Again from [the spec][1]:

> Other specifications may reference this specification by specifying the location of a data file as `$XDG_DATA_DIRS/subdir/filename`. This implies that:
>
> * Such file should be installed to `$datadir/subdir/filename` with `$datadir` defaulting to `/usr/share`.
>
> * A user specific version of the data file may be created in `$XDG_DATA_HOME/subdir/filename`, taking into account the default value for `$XDG_DATA_HOME` if `$XDG_DATA_HOME` is not set.
>
> * Lookups of the data file should search for `./subdir/filename` relative to all base directories specified by `$XDG_DATA_HOME` and `$XDG_DATA_DIRS`. If an environment variable is either not set or empty, its default value as defined by this specification should be used instead.

`containers-storage` would be the subdir specific to the containers/storage library.  I don't see a need to group all of the container data under a `containers/` directory, because it makes managing that shared directory more complicated.  `dataDir` will still be a shared directory potentially consumed by several packages, so sharing directories is obviously something that can work, but I think `containers/storage` is unnecessarily deep.

Finally, the chained-lookup point at the end of the XDG quote above is something that we'd have to approach by patching containers/storage itself.  If we end up patching containers/storage to support the XDG spec (which seems useful to me), we probably won't need *any* code here.  We can just leave `GraphRoot` empty, and let containers/storage pull it in from the XDG variables and its internal `subdir/filename`, etc. defaults.

[1]: https://specifications.freedesktop.org/basedir-spec/basedir-spec-0.7.html
[2]: https://github.com/casimir/xdg-go/blob/372ccc2180dab73316615641d5617c5ed3e35529/xdg.go#L15-L25

Reply:
Lets open a PR for containers/storage and see what @nalind thinks, but for now, we can handle this here, and just get this PR merged, when it is ready.

Reply:
I am in favour of keeping the two levels `containers/storage` as we also have `containers/atomic`

Reply:
> ... but for now, we can handle this here, and just get this PR merged, when it is ready.

The `dataDir` unification in my earlier comment (which DRYs up the current two lines with `"containers", "storage"`) can happen in this PR without blocking on upstream containers/storage work.